orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/non_file_metadata/typing.py

```diff
@@ -0,0 +1,12 @@
+from typing import NamedTuple
+
+from hca_orchestration.support.typing import HcaScratchDatasetName
+
+
+class NonFileMetadataType(str):
```

Comment:
I made these in parallel to what Drew made for the file metadata types, again to keep stuff tightly typed; if we think it makes sense for it all to fall under the same type, I can change it to that as well.